narwhals/dataframe.py

```diff
@@ -2005,19 +1966,25 @@ def columns(self) -> list[str]:
         Get column names.
 
         Examples:
-
+            >>> import pandas as pd
             >>> import polars as pl
             >>> import narwhals as nw
-            >>> lf_pl = pl.LazyFrame(
-            ...     {
-            ...         "foo": [1, 2, 3],
-            ...         "bar": [6, 7, 8],
-            ...         "ham": ["a", "b", "c"],
-            ...     }
-            ... ).select("foo", "bar")
-            >>> lf = nw.from_native(lf_pl)
-            >>> lf.columns  # doctest: +SKIP
-            ['foo', 'bar']
+            >>> df = {"foo": [1, 2, 3], "bar": [6.0, 7.0, 8.0], "ham": ["a", "b", "c"]}
+            >>> df_pd = pd.DataFrame(df)
+            >>> lf_pl = pl.LazyFrame(df)
+
+            We define a library agnostic function:
+
+            >>> @nw.narwhalify
+            ... def func(df):
+            ...     return df.columns
+
+            We can then pass either pandas or Polars to `func`:
+
+            >>> func(df_pd)
+            ['foo', 'bar', 'ham']
+            >>> func(lf_pl)  # doctest: +SKIP
```

Comment:
Related to [comment](https://github.com/narwhals-dev/narwhals/issues/313#issuecomment-2209081112)